File: server/models/Property.js
const mongoose = require("mongoose");

const propertySchema = new mongoose.Schema({
  propertyName: { type: String, required: true },
  description: String,
  location: String,
  pincode: String,

  type: {
    type: String,
    enum: ["sale", "rent", "house", "apartment", "duplex", "land", "commercial"],
    required: true
  },

  status: {
    type: String,
    enum: ["for-sale", "for-rent", "sold", "rented"],
    default: "for-sale"
  },

  price: { type: Number, required: true },

  images: [String], // 🔥 for Cloudinary later

  //house-specific fields
  length: { type: Number },
  width  : { type: Number },
  bedrooms: { type: Number },
  bathrooms: { type: Number },
  totalSqft: { type: Number },

  // residential preference (house / apartment / duplex only)
  vegPreference: {
    type: String,
    enum: ["veg", "non-veg", "any"],
    default: null
  },

  // Owner reference
  owner: { type: mongoose.Schema.Types.ObjectId, ref: 'Owner' }

}, { timestamps: true });

module.exports = mongoose.model("Property", propertySchema);

File: api/models/property.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Property struct {
	ID             primitive.ObjectID   `bson:"_id,omitempty" json:"_id"`
	PropertyName   string               `bson:"propertyName" json:"propertyName"`
	Description    string               `bson:"description" json:"description"`
	Location       string               `bson:"location" json:"location"`
	Pincode        string               `bson:"pincode" json:"pincode"`
	Type           string               `bson:"type" json:"type"`
	Status         string               `bson:"status" json:"status"`
	Price          float64              `bson:"price" json:"price"`
	Images         []string             `bson:"images" json:"images"`
	Length         *float64             `bson:"length,omitempty" json:"length,omitempty"`
	Width          *float64             `bson:"width,omitempty" json:"width,omitempty"`
	Bedrooms       *int                 `bson:"bedrooms,omitempty" json:"bedrooms,omitempty"`
	Bathrooms      *int                 `bson:"bathrooms,omitempty" json:"bathrooms,omitempty"`
	TotalSqft      *float64             `bson:"totalSqft,omitempty" json:"totalSqft,omitempty"`
	VegPreference  *string              `bson:"vegPreference,omitempty" json:"vegPreference,omitempty"`
	Owner          *primitive.ObjectID  `bson:"owner,omitempty" json:"owner,omitempty"`
	OwnerData      *Owner               `bson:"ownerData,omitempty" json:"ownerData,omitempty"`
	CreatedAt      primitive.DateTime   `bson:"createdAt" json:"createdAt"`
	UpdatedAt      primitive.DateTime   `bson:"updatedAt" json:"updatedAt"`
}
